dds_web/web/user.py

```diff
@@ -180,9 +180,15 @@ def confirm_2fa():
     if form.validate_on_submit():
         try:
             token = flask.request.cookies["2fa_initiated_token"]
-            user = dds_web.security.auth.verify_token(token)
-        except (KeyError, ddserr.AuthenticationError):
-            flask.flash("You have to first supply valid credentials", "danger")
+            user = dds_web.security.auth.verify_token_no_data(token)
+        except ddserr.AuthenticationError as e:
+            flask.flash(f"Error: Second factor could not be validated due to: {e}", "danger")
+            return flask.redirect(flask.url_for("auth_blueprint.login", next=next))
+        except Exception as e:
+            flask.current_app.logger.exception(e)
```

Comment:
Do we need to log only this one and not the previous one? Saying just to be sure...

Reply:
The logic was that the general catch all is an unknown error that we would have to debug. The other one should be a known cause. But yeah, I'm not up to date with the latest logging policies so I could change it to whatever really.